src/main/scala/uk/gov/hmrc/play/audit/http/HttpAuditing.scala

```diff
@@ -84,10 +94,90 @@ trait HttpAuditing extends DateTimeUtils {
       hc.names.token -> hc.token.map(_.value).getOrElse("-"),
       Path -> request.url,
       Method -> request.verb) ++
-      request.body.map(b => Seq(RequestBody -> b.toString)).getOrElse(Seq.empty) ++
+      request.body.map(b => Seq(RequestBody -> maskRequestBody(b))).getOrElse(Seq.empty) ++
       HeaderFieldsExtractor.optionalAuditFields(hc.extraHeaders.toMap)
   }
 
+  private def maskRequestBody(body:Any):String = {
+    //The request body comes from calls to executeHooks in http-verbs
+    //It is either called with
+    // - a Map in the case of a web form
+    // - a String created from Json.stringify(wts.writes(... in the case of a class
+    // - a String in the case of a string (where the string can be XML)
+    body match {
+      case m: Map[_, _] => m.map {
+        case (key, _) if maskField(key.asInstanceOf[String]) => (key, MaskValue)
+        case other => other
+      }.toString
+      case s: String => maskString(s)
+      case other => throw new Exception(s"Unexpected type for requestBody when auditing ${outboundCallAuditType}: ${other.getClass}")
+    }
+  }
+
+  private def maskString(text:String) = {
```

Comment:
Do we not have the Content-Type/MIME info guide this?

Reply:
Not this far into the code. I couldn't see it in http-verbs either although I'm trying to keep the change small today.